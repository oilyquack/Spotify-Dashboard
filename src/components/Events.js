import React from "react";
import Event from "./Event";

function Events({ eventList }) {
  return (
    <div className="app__events">
      {eventList
        .sort(function(a, b) {
          return (
            new Date(a.dates.start.localDate) -
            new Date(b.dates.start.localDate)
          );
        })
        .map(function(event) {
          return <Event key={event.id} event={event} />;
        })}
    </div>
  );
}

export default Events;
