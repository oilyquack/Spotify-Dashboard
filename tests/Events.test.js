import React from "react";
import Events from "../src/components/Events";
import renderer from "react-test-renderer";
import { mockEventList } from "./mockEventsResults";

describe("Events", () => {
  it("renders correctly when an eventList is supplied", () => {
    const tree = renderer.create(<Events eventList={mockEventList} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders nothing when the result list is empty", () => {
    const tree = renderer.create(<Events eventList={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
