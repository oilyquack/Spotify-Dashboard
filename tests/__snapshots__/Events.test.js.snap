// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events renders correctly when an eventList is supplied 1`] = `
<div
  className="app__events"
>
  <div
    className="app__events__event"
  >
    <p>
      2018-08-23
    </p>
    <p>
      Sunlight Supply Amphitheater
    </p>
    <p>
      Ridgefield
      ,
       
      United States Of America
    </p>
    <p>
      <a
        href="https://www.ticketmaster.com/slayer-ridgefield-washington-08-23-2018/event/0F00546222D78585"
        target="_blank"
      >
        Get tickets here!
      </a>
    </p>
  </div>
  <div
    className="app__events__event"
  >
    <p>
      2018-08-26
    </p>
    <p>
      SAP Center at San Jose
    </p>
    <p>
      San Jose
      ,
       
      United States Of America
    </p>
    <p>
      <a
        href="https://www.ticketmaster.com/slayer-san-jose-california-08-26-2018/event/1C005461CF80A081"
        target="_blank"
      >
        Get tickets here!
      </a>
    </p>
  </div>
</div>
`;

exports[`Events renders nothing when the result list is empty 1`] = `
<div
  className="app__events"
/>
`;
